pkg/daemon/certs/certs.go

```diff
@@ -25,27 +25,103 @@ import (
 	"encoding/pem"
 	"errors"
 	"fmt"
+	"io"
 	"os"
+	"reflect"
 	"syscall"
 	"unsafe"
 
 	"golang.org/x/sys/windows"
 	"k8s.io/klog/v2"
 )
 
-// Reconcile ensures the certificates in the given trusted CA bundle file are imported as system certificates.
+// importedCABundleFile tracks all the certs that the operator has imported into the node's local trust store
+const importedCABundleFile = "C:\\Temp\\imported-certs.pem"
+
+// Reconcile ensures the node's system certificates match the expected state as given by the trusted CA bundle file.
 // Returns a boolean whether any certs needed to be reconciled
-func Reconcile(caBundle string) (bool, error) {
-	if caBundle == "" {
-		// TODO: ensure that all certs that we've added previously are removed https://issues.redhat.com/browse/WINC-688
+func Reconcile(caBundlePath string) (bool, error) {
+	// caBundlePath will be empty when a cluster-wide proxy is being removed or not in use
+	certChange, err := reconcileCerts(caBundlePath)
+	if err != nil {
+		return false, err
+	}
+	// An error updating the imported bundle file will be self-corrected on the next reconcile, since it is the source
+	// of truth for determining whether certs need to be updated
+	return certChange, updateImportedCABundle(caBundlePath, certChange)
```

Comment:
What if this errors out after doing the actual cert changes?
It won't restart because at the callsites of Reconcile:
```
	certsRemoved, err := certs.Reconcile("")
	if err != nil {
		return err
	}
```
a restart only happens when no error occurs.

Then on the next run, there will be no cert changes as they were already made.
A restart wont be triggered.
In this case a restart is never triggered and things won't function properly.
I suspect you need to restart even in the case of error, if the returned bool is true. This should be documented in the function doc as well.

Reply:
You've convinced me this is a valid concern. doc'd and changed call sites